ShaellLang/SProcess.cs

```diff
@@ -8,40 +8,76 @@ namespace ShaellLang;
 
 public class SProcess : BaseValue, IFunction
 {
-    private Process _process = new Process();
+    public Process Process = new Process();
+    public SProcess LeftProcess;
+    public string Stdin = null;
+    public bool Executed { get; private set; }
     public SProcess(string file) 
         : base("process")
     {
-        _process.StartInfo.FileName = file;
+        Process.StartInfo.FileName = file;
     }
 
-    private void AddArguments(IEnumerable<IValue> args)
+    public void AddArguments(IEnumerable<IValue> args)
     {
         foreach (var arg in args)
         {
             AddArg(arg.ToSString().Val);
         }
     }
 
-    private void AddArg(string str) => _process.StartInfo.ArgumentList.Add(str);
-    public void Dispose() => _process.Dispose();
+    public void AddPipeProcess(SProcess process)
+    {
+        LeftProcess = process;
+    }
 
-    private JobObject Run(Process process)
+    private void AddArg(string str) => Process.StartInfo.ArgumentList.Add(str);
+    public void Dispose() => Process.Dispose();
+
+    private JobObject Run(Process process, string stdin)
     {
-        return JobObject.Factory.StartProcess(process);
+        if (!Executed)
+        {
+            Executed = true;
+            return JobObject.Factory.StartProcess(process, stdin);
+        }
+
+        return null;
     }
 
     public IValue Call(IEnumerable<IValue> args)
     {
         AddArguments(args);
-        return Run(_process);
+        return Run(Process, Stdin);
+    }
+
+    public JobObject Execute(SProcess parentProc)
```

Comment:
Synes at der skal `Execute()` overload uden nogen argumenter istedet for at give `null`